### DISCUSSION TITLE
truffleruby+graalvm-dev fails to download for install

### DISCUSSION THREAD

**Discussion by halostatue** [rbenv/ruby-build#2242] [Build failures] [answered]
```console
$ ruby-build truffleruby+graalvm-dev ~/.rubies/truffleruby+graalvm-dev
To follow progress, use 'tail -f /var/folders/c6/k_j3j3w11590l8x6t8rwxw200000gn/T/ruby-build.20230901120615.95886.log' or pass --verbose
ruby-build: using openssl from homebrew
Downloading graalvm-community-java17-darwin-aarch64-dev.tar.gz...
-> https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java17-darwin-aarch64-dev.tar.gz
error: failed to download truffleruby+graalvm-dev.tar.gz
```

This is affecting any GitHub Action which is using `truffleruby+graalvm-head` (which uses `truffleruby+graalvm-dev`):

```
  /home/runner/work/_temp/ruby-build-for-setup-ruby/bin/ruby-build truffleruby+graalvm-dev /home/runner/.rubies/truffleruby+graalvm-head
  To follow progress, use 'tail -f /tmp/ruby-build.20230901153346.1773.log' or pass --verbose
  Downloading graalvm-community-java17-linux-amd64-dev.tar.gz...
  -> https://github.com/graalvm/graalvm-ce-dev-builds/releases/latest/download/graalvm-community-java17-linux-amd64-dev.tar.gz
  error: failed to download truffleruby+graalvm-dev.tar.gz
  
  BUILD FAILED (Ubuntu 20.04 using ruby-build 20230717-11-g4d4678b)
```

See https://github.com/KineticCafe/app_identity/actions/runs/6051304100/job/16422454037?pr=69

I have not opened an issue with ruby/setup-ruby because this is reproducible with ruby-build on my macOS, not just the GitHub Action runner using Ubuntu 20.04 or 22.04.

**Comment by andrykonchin** [comment]
Probably `java21` should be used instead of `java17` in https://github.com/rbenv/ruby-build/blob/master/share/ruby-build/truffleruby%2Bgraalvm-dev.

There are only the following assets versions available:

- [graalvm-community-java21-darwin-aarch64-dev.tar.gz](https://github.com/graalvm/graalvm-ce-dev-builds/releases/download/23.1.0-dev-20230817_2024/graalvm-community-java21-darwin-aarch64-dev.tar.gz)
- [graalvm-community-java21-darwin-amd64-dev.tar.gz](https://github.com/graalvm/graalvm-ce-dev-builds/releases/download/23.1.0-dev-20230817_2024/graalvm-community-java21-darwin-amd64-dev.tar.gz)
- [graalvm-community-java21-linux-aarch64-dev.tar.gz](https://github.com/graalvm/graalvm-ce-dev-builds/releases/download/23.1.0-dev-20230817_2024/graalvm-community-java21-linux-aarch64-dev.tar.gz)
- [graalvm-community-java21-linux-amd64-dev.tar.gz](https://github.com/graalvm/graalvm-ce-dev-builds/releases/download/23.1.0-dev-20230817_2024/graalvm-community-java21-linux-amd64-dev.tar.gz)
- [graalvm-community-java21-windows-amd64-dev.zip](https://github.com/graalvm/graalvm-ce-dev-builds/releases/download/23.1.0-dev-20230817_2024/graalvm-community-java21-windows-amd64-dev.zip)

**Reply by eregon** [reply]
Right, I opened https://github.com/rbenv/ruby-build/issues/2243 and will look into it.

**Reply by eregon** [reply, marked as answer]
Fixed in https://github.com/rbenv/ruby-build/pull/2244

**Comment by eregon** [comment]
@halostatue Just curious, is there a reason to use `truffleruby+graalvm` and not just `truffleruby`?
That would use the native TruffleRuby standalone, which is typically better for CI (unless one needs interop with JS or wants to run in --jvm mode).

**Reply by eregon** [reply]
I see you test both actually. I think it would be fine to remove `truffleruby+graalvm`, testing against `truffleruby` is enough.

**Reply by halostatue** [reply]
OK. It’ll take me some time to go through all of my different actions, but I can do that.
